### PR TITLE
TDD

### PR DESCRIPTION
**Pull request by rivera-fernando**

### CODE REVIEW

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -14,10 +14,121 @@
...
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+
+      Collection<String> attendees = request.getAttendees();
+      Collection<String> optional = request.getOptionalAttendees();
+      long duration = request.getDuration();
+      //EDGE CASES
+      //return whole day if the meeting to schedule requires no attendees
```
Comment (serzhk) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:30]: That's sounds illogically :) If there is no attendees - no meetings should be scheduled?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -14,10 +14,121 @@
...
+  public Collection<TimeRange> getAvailableTimes(ArrayList<Event> merged_events, MeetingRequest request) {
+      Collection<TimeRange> potential = new ArrayList<>();
+      Collection<TimeRange> finalized = new ArrayList<>();
+      if (merged_events.size() == 1) {
+        potential.add(TimeRange.fromStartEnd(0, merged_events.get(0).getWhen().start(), false));
+        potential.add(TimeRange.fromStartEnd(merged_events.get(0).getWhen().end(), 24*60, false));
+      }
+      else {
+      
+      for (int i = 0; i < merged_events.size(); i++) {
```
Comment (serzhk) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: indentation

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 3]
```diff
@@ -14,10 +14,121 @@
...
+      Collection<String> attendees = request.getAttendees();
+      Collection<String> optional = request.getOptionalAttendees();
+      long duration = request.getDuration();
+      //EDGE CASES
+      //return whole day if the meeting to schedule requires no attendees
+      if (attendees.size() == 0 && optional.size() == 0) {
+          return Arrays.asList(TimeRange.WHOLE_DAY);
+      }
+      //return no availability if the meeting to schedule is longer than the day
+      else if (duration > 24*60) {
```
Comment (wulian1223) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: `if` should be good enough since you `return` in every condition.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (wulian1223) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:30]: +1

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java [hunk 4]
```diff
@@ -270,5 +359,64 @@ public void notEnoughRoom() {
...
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected = Arrays.asList();
+
+    Assert.assertEquals(expected, actual);
+
+  }
 }
 
+
```
Comment (wulian1223) [inline walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:422]: test when meeting to schedule is longer than the day?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (rivera-fernando) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:30]: I wrote this edge case based on the test provided for us
```
@Test
  public void optionsForNoAttendees() {
    MeetingRequest request = new MeetingRequest(NO_ATTENDEES, DURATION_1_HOUR);

    Collection<TimeRange> actual = query.query(NO_EVENTS, request);
    Collection<TimeRange> expected = Arrays.asList(TimeRange.WHOLE_DAY);

    Assert.assertEquals(expected, actual);
  }
```
did I misunderstand the desired functionality?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 2 above)
Comment (rivera-fernando) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Done.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 3 above)
Comment (rivera-fernando) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Done, and fixed corresponding indentation issues.

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java (hunk 4 above)
Comment (rivera-fernando) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:422]: I believe that is this test that was provided for us
```
@Test
  public void noOptionsForTooLongOfARequest() {
    // The duration should be longer than a day. This means there should be no options.
    int duration = TimeRange.WHOLE_DAY.duration() + 1;
    MeetingRequest request = new MeetingRequest(Arrays.asList(PERSON_A), duration);

    Collection<TimeRange> actual = query.query(NO_EVENTS, request);
    Collection<TimeRange> expected = Arrays.asList();

    Assert.assertEquals(expected, actual);
  }
```
unless you are referencing something else that I am missing?

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (serzhk) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java:30]: done

File: walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java (hunk 4 above)
Comment (serzhk) [reply walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java:422]: done

**Review by serzhk — APPROVED** [review]
